### PR TITLE
Get user orders by user email instead of rely on current user

### PR DESCRIPTION
**Pull request by mitogh**
The problem was because `is_user_logged_in` it was not working 100% all the
times and the:
- POST
- GET

Endpoints for `create-order` was not working as expected, in this case a change
to use the user email is a more consistent value same as it's working with the
`token_id`.

This change also fix a bug where with any random token we were able to get all
the orders from all the anonymous users.

### CODE REVIEW

File: src/Api/Order.php [hunk 1]
```diff
@@ -92,6 +92,13 @@ public function endpoint_args() {
 					return false === $token_id || is_string( $token_id );
 				},
 			],
+			'user_email' => [
```
Comment (Rulox) [inline src/Api/Order.php:95]: So, does this mean, that I can get the information of anyone if I know his email? @mitogh

**Review by Rulox — COMMENTED** [review]
Did you see my comment here @mitogh ? I just wanted to understand if that how it works! just to be sure it's secure

File: src/Api/Order.php (hunk 1 above)
Comment (mitogh) [reply src/Api/Order.php:95]: That's correct, but the information of the order, it works in the same way as the token.

We can use the user ID but I think that's more risky to use. Let me know what do you think, thanks.

Comment (Rulox) [reply src/Api/Order.php:95]: Ok let's go with this then!

**Pull request merged**